it/controllers/benefits/CompanyVanFuelBenefitsControllerISpec.scala

```diff
@@ -0,0 +1,429 @@
+
+            "has an OK status" in {
+              result.status shouldBe OK
+            }
+
+            titleCheck(user.specificExpectedResults.get.expectedTitle)
+            h1Check(user.specificExpectedResults.get.expectedH1)
+            textOnPageCheck(expectedCaption(taxYearEOY), captionSelector)
+            radioButtonCheck(yesText, 1, Some(true))
+            radioButtonCheck(noText, 2, Some(false))
+            buttonCheck(expectedButtonText, continueButtonSelector)
+            formPostLinkCheck(continueLink, continueButtonFormSelector)
+            welshToggleCheck(user.isWelsh)
+          }
+          "Render the 'Did you receive van fuel benefits' page with the correct content with yes value pre-filled and van fuel amount not answered" which {
+            lazy val result: WSResponse = {
+              dropEmploymentDB()
+              userDataStub(userData(fullEmploymentsModel()), nino, taxYearEOY)
+              insertCyaData(cya(isPriorSubmission = false, benefitsWithVanFuelYes(vanFuelAmount = None)), userRequest)
+              authoriseAgentOrIndividual(user.isAgent)
+              urlGet(vanFuelBenefitsPage(taxYearEOY), welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+            }
+
+            implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+            import Selectors._
+            import user.commonExpectedResults._
+
+            "has an OK status" in {
+              result.status shouldBe OK
+            }
+
+            titleCheck(user.specificExpectedResults.get.expectedTitle)
+            h1Check(user.specificExpectedResults.get.expectedH1)
+            textOnPageCheck(expectedCaption(taxYearEOY), captionSelector)
+            radioButtonCheck(yesText, 1, Some(true))
+            radioButtonCheck(noText, 2, Some(false))
+            buttonCheck(expectedButtonText, continueButtonSelector)
+            formPostLinkCheck(continueLink, continueButtonFormSelector)
+            welshToggleCheck(user.isWelsh)
+          }
+
+          /*
+          Tests for redirects
+                  Redirect to benefits CYA when:
+                  - if van question is false
+                  - van section is not finished (vanQuestion and van)
+
+           */
+
+          "Redirect user to the check your benefits page" which {
+            lazy val result: WSResponse = {
+              dropEmploymentDB()
+              authoriseAgentOrIndividual(user.isAgent)
+              urlGet(vanFuelBenefitsPage(taxYearEOY), user.isWelsh, follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+            }
+
+            "has an SEE_OTHER(303) status" in {
+              result.status shouldBe SEE_OTHER
+              result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+            }
+          }
+        }
+      }
+    }
+
+
+    //  ".submit" should {
```

Comment:
You have no submit tests (i.e post tests when the user submits.) They are commented out
Refer to the examples ISPEC I sent as already fixes all this.